### PR TITLE
fix: add default route on customer vnets and provide default route from CNS to CNI

### PR DESCRIPTION
**Pull request by paulyufan2**
**Reason for Change**:

This PR is to fix one issue in swiftv2 Windows scenario:
When a pod is created, the default route is added on infra vnet:
root@swiftv2-pod-3:/# ip route
**default via 10.244.2.1 dev eth0 metric 1**
It leads to ping a VM IP in the same VNET that cannot work.

There are two issues::
1.CNS does not provide the default route to CNI;
2.CNI should only add the default route to secondary interface customer vnet; on Swiftv2 scenario, skipDefaultRoutes is set to true for infraNIC interface and false for a secondary interface; so if !info.skipDefaultRoutes, then add default route.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] relevant PR labels added

**Notes**:

### CODE REVIEW

**Review by tamilmani1989 — CHANGES_REQUESTED** [review]
how it works for linux not windows? why we need this change specifically for windows

File: cni/network/invoker_cns.go [hunk 1]
```diff
@@ -439,11 +439,15 @@ func configureDefaultAddResult(info *IPResultInfo, addConfig *IPAMAddConfig, add
...
-				Dst: defaultRouteDstPrefix,
-				Gw:  ncgw,
-			})
+			// on Swiftv2 windows scenario, skip to add default route on infra vnet;
+			if !info.skipDefaultRoutes {
+				resRoute = append(resRoute, network.RouteInfo{
+					Dst: defaultRouteDstPrefix,
+					Gw:  ncgw,
+				})
+			}
```
Comment (tamilmani1989) [inline cni/network/invoker_cns.go:448]: if routes are specified as part of cns goalstate why cni need to construct new one?

Comment (tamilmani1989) [reply cni/network/invoker_cns.go:448]: what scenario this else case solves?

File: network/endpoint_windows.go [hunk 2]
```diff
@@ -428,7 +428,7 @@ func (nw *network) newEndpointImplHnsV2(cli apipaClient, epInfo *EndpointInfo) (
 	}
 
 	// Create the HCN endpoint.
-	logger.Info("Creating hcn endpoint", zap.Any("hcnEndpoint", hcnEndpoint), zap.String("computenetwork", hcnEndpoint.HostComputeNetwork))
+	logger.Info("Creating hcn endpoint", zap.Any("hcnEndpoint", hcnEndpoint), zap.String("computenetwork", hcnEndpoint.HostComputeNetwork), zap.Any("routes", hcnEndpoint.Routes))
```
Comment (tamilmani1989) [inline network/endpoint_windows.go:431]: can skip logging routes if hcnendpoint object already logs it.

**Review by tamilmani1989 — CHANGES_REQUESTED** [review]

File: network/endpoint_windows.go (hunk 2 above)
Comment (paulyufan2) [reply network/endpoint_windows.go:431]: zap log does not log Routes field from hcnEndpoint

File: cni/network/invoker_cns.go (hunk 1 above)
Comment (paulyufan2) [reply cni/network/invoker_cns.go]: it's not correct...I built image based on this code and did not push the correct codes.

File: cns/middlewares/k8sSwiftV2_windows.go [hunk 3]
```diff
@@ -40,5 +40,12 @@ func (k *K8sSWIFTv2Middleware) assignSubnetPrefixLengthFields(podIPInfo *cns.Pod
 		},
 		GatewayIPAddress: interfaceInfo.GatewayIP,
 	}
+	// assign default route
+	route := cns.Route{
+		IPAddress:        "0.0.0.0/0",
```
Comment (prmathur-microsoft) [inline cns/middlewares/k8sSwiftV2_windows.go:45]: Why 0.0.0.0/0

Comment (kmurudi) [reply cns/middlewares/k8sSwiftV2_windows.go:45]: is this added for linux path though in current scenario?

Comment (paulyufan2) [reply cns/middlewares/k8sSwiftV2_windows.go:45]: yes, it's added for Linux path, we should do the same thing for windows path

**Comment by paulyufan2** [comment]
will use another PR for long-term solution

**Pull request closed without merging**